src/lib/utils/socket/socket.cpp

```diff
@@ -46,20 +46,19 @@ class Asio_Socket final : public OS::Socket {
    public:
       Asio_Socket(std::string_view hostname, std::string_view service, std::chrono::milliseconds timeout) :
             m_timeout(timeout), m_timer(m_io), m_tcp(m_io) {
-         m_timer.expires_from_now(m_timeout);
+         m_timer.expires_after(m_timeout);
          check_timeout();
 
          boost::asio::ip::tcp::resolver resolver(m_io);
-         boost::asio::ip::tcp::resolver::query query(std::string{hostname}, std::string{service});
-         boost::asio::ip::tcp::resolver::iterator dns_iter = resolver.resolve(query);
+         boost::asio::ip::tcp::resolver::results_type dns_iter =
+            resolver.resolve(std::string{hostname}, std::string{service});
 
          boost::system::error_code ec = boost::asio::error::would_block;
 
-         auto connect_cb = [&ec](const boost::system::error_code& e, const boost::asio::ip::tcp::resolver::iterator&) {
-            ec = e;
-         };
+         auto connect_cb = [&ec](const boost::system::error_code& e,
+                                 const boost::asio::ip::tcp::resolver::results_type::iterator&) { ec = e; };
```

Comment:
Another nit: If we don't care about the second parameter anyway, we might as well `auto` it:

```suggestion
         auto connect_cb = [&ec](const boost::system::error_code& e,
                                 const auto&) { ec = e; };
```